adhocracy4/comments_async/static/comments_async/comment_box.jsx

```diff
@@ -209,12 +215,16 @@ export default class CommentBox extends React.Component {
       comment = comments[parentIndex].child_comments[index]
     }
 
+    // flushSync stops react18 batching state updates and ensures
+    // re-render when no error
     return api.comments.change(modifiedComment, comment.id)
       .done(changed => {
-        this.updateStateComment(
-          index,
-          parentIndex,
-          changed)
+        flushSync(() => {
```

Comment:
Do we really need this? is the automatic batching problematic here?

Reply:
yeah, it's what I spent a day trying to fix :p without it it doesn't re-render when you edit comments or categories. 
Other option is instead of having 2 set states we combine them, not sure what happens then with errors and re-render but could test. Also thought having this was a good reminder about the batching behaviour